lrpar/src/lib/builder.rs

```diff
@@ -235,26 +238,33 @@ pub fn parse(lexemes: &[Lexeme<{storaget}>])
         }
 
         // Output the cache so that we can check whether the IDs map is stable.
-        outs.push_str(&imc);
+        outs.push_str(&cache);
 
         let mut f = File::create(outp_rs)?;
         f.write_all(outs.as_bytes())?;
         Ok(rule_ids)
     }
 
-    /// Generate the rule IDs map cache. We don't need to be particularly clever here: we just need
-    /// to record the identifiers and nonterminal names so that we can check if they've changed
-    /// later.
-    fn gen_ids_map_cache(&self, grm: &YaccGrammar<StorageT>) -> String {
+    /// Generate the cache, which determines if anything's changed enough that we need to
+    /// regenerate outputs and force rustc to recompile.
+    fn rebuild_cache(&self, grm: &YaccGrammar<StorageT>) -> String {
+        // We don't need to be particularly clever here: we just need to record the various things
+        // that could change between builds.
         let mut cache = String::new();
         cache.push_str(" \n/* CACHE INFORMATION\n");
+
```

Comment:
I'm not sure that you really need these extra spaces, especially when there are already comments to break up the code.

Reply:
They're to break up the different parts of the rebuild cache. I think whoever comes along in the future might be grateful for the visual clues. [Given that I got this bit wrong, and I'm the idiot who wrote it, I'm assuming it'll be harder for the people who come after me ;)]

Reply:
Do you not think the comments are sufficient for that? Or could the comments be improved?

Reply:
Whitespace inside functions is like a paragraph break to me; functions are sections. I wouldn't want to read sections in papers without paragraphs!